### ISSUE TITLE
Develop a re-usable installation method for the SNAP toolbox

### ISSUE THREAD

**Issue by agstephens** [cedadev/extra-sci-packages#13]
The SNAP toolbox is a collection of tools used for working with ESA satellite data, such as those from the Sentinel missions.

We would like to **install SNAP under `/apps/jasmin/supported/`** (as long as it is not too difficult to install/maintain). We probably do not want to install it as part of `extra-sci-packages` unless there is a good reason to do so.

Here is the download location for the installer: 

http://step.esa.int/main/download/snap-download/

This is a little basic install guide: 

https://www.gis-blog.com/how-to-install-esa-snap-on-ubuntu-linux/

Many users will want to use the python API, `snappy`. Instructions for setting up the SNAP python API are here: 

https://senbox.atlassian.net/wiki/spaces/SNAP/pages/19300362/How+to+use+the+SNAP+API+from+Python

Apparently, multiple users will need to be able to _hook_ into the install and point towards it with an environment variable when loading the `snappy` module. ESA likely has an internal answer to this problem, so if we run into problems we should get in touch with them.

The task is:
- [x] Install the SNAP Toolbox
- [x] Get it running with the Python API
- [x] Write some brief tests to check it works
- [x] Create a re-usable recipe to do the installation into: `/apps/jasmin/supported/snap`
- [x] Create a module file for this under: `/apps/jasmin/modulefiles/`

@alaniwi - **NOTE: If you have spent more than 2 days on this then leave it to discuss further when @agstephens  gets back from leave.**

**Comment by alaniwi** [comment]
It is installed and I have written a detailed document explaining how to repeat the installation on JASMIN. The document also explains why a scripted install is not really possible.

See `/apps/jasmin/supported/snap/INSTALLATION/README-snap-on-jasmin`

It will run better when the `mesa-dri-drivers` RPM is provided on the sci machines (as seen on `sci2-test` where I've installed it), and I've created a separate ticket to add the relevant meta RPM.

Part of the build process needs to be on test machine because the production sci machines don't have JDK in the OS -- they have JRE.  Maybe we should provide the corresponding JDK RPM also via extra-sci-packages.  Though there is also an argument for providing Java via Jaspy.
